apis/externalsecrets/v1/externalsecret_types.go

```diff
@@ -181,6 +184,21 @@ type TemplateRefItem struct {
 	TemplateAs TemplateScope `json:"templateAs,omitempty"`
 }
 
+// ManifestTarget defines a custom Kubernetes resource type to be created
+// instead of a Secret. This allows ExternalSecret to create ConfigMaps,
+// Custom Resources, or any other Kubernetes resource type.
+type ManifestTarget struct {
```

Comment:
nit: This should be called reference in alignment with existing k8s reference concepts, so `ManifestReference`